e2e/helm_client.go

```diff
@@ -0,0 +1,286 @@
+	stdout, err := m.executeCommandWithValues(chartName, "install", values)
+	if err != nil {
+		return "", err
+	}
+	return getNameFromHelmOutput(stdout), nil
+}
+
+// Status reports nil if release is considered to be succesfull
+func (m *BinaryHelmManager) Status(releaseName string) error {
+	stdout, err := m.executeUsingHelm("status", releaseName, "--tiller-namespace", m.Namespace)
+	if err != nil {
+		return err
+	}
+	status := getStatusFromHelmOutput(stdout)
+	if status == "DEPLOYED" {
+		return nil
+	}
+	return fmt.Errorf("Expected status is DEPLOYED. But got %v for release %v.", status, releaseName)
+}
+
+func (m *BinaryHelmManager) Delete(releaseName string) error {
+	_, err := m.executeUsingHelm("delete", releaseName, "--tiller-namespace", m.Namespace)
+	return err
+}
+
+func (m *BinaryHelmManager) Upgrade(chartName, releaseName string, values map[string]string) error {
+	arg := make([]string, 0, 9)
+	arg = append(arg, "upgrade", releaseName, chartName)
+	if len(values) > 0 {
+		arg = append(arg, "--set", prepareArgsFromValues(values))
+	}
+	_, err := m.executeUsingHelmInNamespace(arg...)
+	return err
+}
+
+func (m *BinaryHelmManager) Rollback(releaseName string, revision int) error {
+	arg := make([]string, 0, 6)
+	arg = append(arg, "rollback", releaseName, strconv.Itoa(revision), "--tiller-namespace", m.Namespace)
+	_, err := m.executeUsingHelm(arg...)
+	return err
+}
+
+func (m *BinaryHelmManager) executeUsingHelmInNamespace(arg ...string) (string, error) {
+	arg = append(arg, "--namespace", m.Namespace, "--tiller-namespace", m.Namespace)
+	return m.executeUsingHelm(arg...)
+}
+
+func (m *BinaryHelmManager) executeUsingHelm(arg ...string) (string, error) {
+	if m.TillerHost != "" {
+		arg = append(arg, "--host", m.TillerHost)
+	}
+	return m.executeUsingBinary(m.HelmBin, arg...)
+}
+
+func (m *BinaryHelmManager) executeUsingBinary(binary string, arg ...string) (string, error) {
+	cmd := exec.Command(binary, arg...)
+	Logf("Running command %+v\n", cmd.Args)
+	stdout, err := cmd.Output()
+	if err != nil {
+		switch err.(type) {
+		case *exec.ExitError:
+			stderr := err.(*exec.ExitError)
+			Logf("Command %+v, Err %s\n", cmd.Args, stderr.Stderr)
+		case *exec.Error:
+			Logf("Command %+v, Err %s\n", cmd.Args, err)
+		}
+		return "", err
+	}
+	return string(stdout), nil
+}
+
+func (m *BinaryHelmManager) executeCommandWithValues(releaseName, command string, values map[string]string) (string, error) {
+	arg := make([]string, 0, 8)
+	arg = append(arg, command, releaseName)
+	if len(values) > 0 {
+		var b bytes.Buffer
+		for key, val := range values {
+			b.WriteString(key)
+			b.WriteString("=")
+			b.WriteString(val)
+			b.WriteString(",")
+		}
+		arg = append(arg, "--set", b.String())
+	}
+	return m.executeUsingHelmInNamespace(arg...)
+}
+
+func (m *BinaryHelmManager) InstallServiceAccounts() error {
+	objects := strings.Replace(serviceAccountTemplate, "TILLER_NAMESPACE", m.Namespace, -1)
+
+	f, err := ioutil.TempFile("", m.Namespace)
```

Comment:
Is there any need to close and clean up the file once the execute is done?

Reply:
No, https://golang.org/pkg/io/ioutil/#TempFile claims that `Multiple programs calling TempFile simultaneously will not choose the same file.`. Since e2e tests will mostly run on circle, we can just leave it there to be scrapped with the rest of the build.